src/cursor/cur_file.c

```diff
@@ -608,30 +608,26 @@ __curfile_reopen_int(WT_CURSOR *cursor)
  *     WT_CURSOR->reopen method for the btree cursor type.
  */
 static int
-__curfile_reopen(WT_CURSOR *cursor, bool check_only)
+__curfile_reopen(WT_CURSOR *cursor, bool sweep_check_only)
 {
     WT_DATA_HANDLE *dhandle;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
-    bool can_reopen;
+    bool can_sweep;
 
     session = CUR2S(cursor);
     dhandle = ((WT_CURSOR_BTREE *)cursor)->dhandle;
 
-    if (check_only) {
-        can_reopen = WT_DHANDLE_CAN_REOPEN(dhandle);
+    if (sweep_check_only) {
         /*
-         * If the data handle can't be reopened, it is a candidate for sweep, and that should never
-         * happen if any session (including ours) is actively using it.
+         * The sweep check returns WT_NOTFOUND if the cursor should be swept. Generally if the
+         * associated data handle cannot be reopened we want to sweep it. But we also don't want to
+         * sweep a handle that we're currently operating on. That can occur if we're in the middle
+         * of a close on a cursor for a handle that is marked as dropped, as session cursor sweeps
+         * can happen while closing.
```

Comment:
I would rephrase some of this comment because I'm getting confused about who "we" means. Here's my attempt at understanding in explicit words. It may not be quite what you want and it may not be accurate. Please fix accordingly.

The sweep check returns WT_NOTFOUND if the cursor should be swept. Generally if the
associated data handle cannot be reopened it should be swept. But a handle being operated
on by this thread should not be swept. The situation where a handle cannot be reopened but
also cannot be swept can occur if this thread is in the middle
of closing a cursor for a handle that is marked as dropped at the same time the connection sweep
thread runs and tries to sweep session cursors.

Reply:
I'm using your wording, except for the very end.  The session sweep is actually running here.  The call stack is roughly:
```
    SESSION->close()  calls...
    release cursor.      calls....
    run a few iterations of the session sweep
    call cursor->reopen (__curfile_reopen) to see if we can sweep this cursor
```

Reply:
Alternate wording:
```
        /*
         * The sweep check returns WT_NOTFOUND if the cursor should be swept. Generally if the
         * associated data handle cannot be reopened it should be swept. But a handle being operated
         * on by this thread should not be swept. The situation where a handle cannot be reopened
         * but also cannot be swept can occur if this thread is in the middle of closing a cursor
         * for a handle that is marked as dropped. During the close, a few iterations of
         * the session cursor sweep are run.  The sweep calls this function to see if a cursor
         * should be swept, and it may thus be asking about the very cursor being closed.
         */
```